src/test/java/reposense/parser/ArgsParserTest.java

```diff
@@ -0,0 +1,175 @@
+    private static final String OUTPUT_DIRECTORY_ABSOLUTE = Paths.get(PROJECT_DIRECTORY, "build").toString();
+    private static final String CONFIG_FILE_RELATIVE = Paths.get("sample_small.csv").toString();
+    private static final String OUTPUT_DIRECTORY_RELATIVE = Paths.get("build").toString();
+    private static final String DEFAULT_MANDATORY_ARGS = "-config " + CONFIG_FILE_ABSOLUTE + " ";
+
+    @Test
+    public void parse_allCorrectInputs_success() throws ParseException, IOException {
+        String input = String.format("-config %s -output %s -since 01/07/2017 -until 30/11/2017",
+                CONFIG_FILE_ABSOLUTE, OUTPUT_DIRECTORY_ABSOLUTE);
+        CliArguments cliArguments = ArgsParser.parse(input.split(INPUT_DELIMITER));
+        Files.isSameFile(Paths.get(CONFIG_FILE_ABSOLUTE), cliArguments.getConfigFilePath());
+        Files.isSameFile(Paths.get(OUTPUT_DIRECTORY_ABSOLUTE), cliArguments.getOutputFilePath());
+
+        Date expectedSinceDate = TestUtil.getDate(2017, Calendar.JULY, 1);
+        Date expectedUntilDate = TestUtil.getDate(2017, Calendar.NOVEMBER, 30);
+        Assert.assertEquals(expectedSinceDate, cliArguments.getSinceDate().get());
+        Assert.assertEquals(expectedUntilDate, cliArguments.getUntilDate().get());
+    }
+
+    @Test
+    public void parse_configFileOnly_success() throws ParseException, IOException {
+        String input = String.format("-config %s", CONFIG_FILE_ABSOLUTE);
+        CliArguments cliArgumentsOfAbsolutePath = ArgsParser.parse(input.split(INPUT_DELIMITER));
+
+        input = String.format("-config %s", CONFIG_FILE_RELATIVE);
+        CliArguments cliArgumentsOfRelativePath = ArgsParser.parse(input.split(INPUT_DELIMITER));
+
+        Path expected = Paths.get(CONFIG_FILE_ABSOLUTE);
+        Files.isSameFile(expected, cliArgumentsOfAbsolutePath.getConfigFilePath());
+        Files.isSameFile(expected, cliArgumentsOfRelativePath.getConfigFilePath());
+
+        //Optional arguments have default values
+        Assert.assertEquals(Optional.empty(), cliArgumentsOfAbsolutePath.getSinceDate());
+        Assert.assertEquals(Optional.empty(), cliArgumentsOfAbsolutePath.getUntilDate());
+        Assert.assertTrue(Files.isSameFile(Paths.get("."), cliArgumentsOfAbsolutePath.getOutputFilePath()));
+
+        Assert.assertEquals(Optional.empty(), cliArgumentsOfRelativePath.getSinceDate());
+        Assert.assertEquals(Optional.empty(), cliArgumentsOfRelativePath.getUntilDate());
+        Assert.assertTrue(Files.isSameFile(Paths.get("."), cliArgumentsOfRelativePath.getOutputFilePath()));
+    }
+
+    @Test
+    public void parse_configFileAndOutputDirectory_success() throws ParseException, IOException {
+        String input = String.format("-config %s -output %s", CONFIG_FILE_ABSOLUTE, OUTPUT_DIRECTORY_RELATIVE);
+        CliArguments cliArgAbsConfigRelOutput = ArgsParser.parse(input.split(INPUT_DELIMITER));
```

Comment:
Have you explore such tools as mentioned in this SO? https://stackoverflow.com/questions/3259143/split-a-string-containing-command-line-parameters-into-a-string-in-java

Reply:
Yes... but this prevent the mistake where developers might accidentally type an additional space in the input line and try to find the error all over the place.

Reply:
Also, it will be nice to keep the test arguments as close as possible to what user might type.

Reply:
Also, `cliArgAbsConfigRelOutput` is too complicated. Your first line is explanatory enough. 
Just name it as usual, otherwise you'll have to make all the others occurrence follow this standard too.